### PR TITLE
fix 954, trim whitespace and newline char for range input

### PR DESCRIPTION
**Pull request by wzhanw**
**Which issue is resolved by this Pull Request:** 
Resolves # https://github.com/kubeflow/kfp-tekton/issues/954

**Description of your changes:**
1. Trim space and newline char for range input and validate it
2. Allow whitespace as iteration string.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by wzhanw** [comment]
@Tomcli @ScrapCodes could you review and approve this? we required this fix asap.

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go [hunk 1]
```diff
@@ -1191,6 +1225,42 @@ var expectedPipelineRunIteration1 = &v1beta1.PipelineRun{
 	},
 }
 
+var expectedPipelineRunIteration1WithWhiteSpace = &v1beta1.PipelineRun{
```
Comment (ScrapCodes) [inline tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:1228]: `expectedPipelineRunIteration1WithWhiteSpace` why is this called `expectedPipelineRunIteration1` i.e. 1 at the end?

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go [hunk 2]
```diff
@@ -815,21 +825,26 @@ func computeIterations(run *v1alpha1.Run, tls *pipelineloopv1alpha1.PipelineLoop
 	iterationElements := []interface{}{}
 	iterationParamStr := ""
 	iterationParamStrSeparator := ""
+	var err error
 	for _, p := range run.Spec.Params {
 		if p.Name == "from" {
-			from, _ = strconv.Atoi(p.Value.StringVal)
-			fromProvided = true
+			from, err = getIntegerParamValue(p)
```
Comment (ScrapCodes) [inline tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:831]: In practice, I have not seen a problem due to "\n" terminated strings passed as a value. My understanding was these are already taken care of by the parsers. 

Can you provide an example ?

Comment (wzhanw) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:831]: @ScrapCodes , we have an example, the value of the range is from an upper task result,  e.g: an output of a bash script task.  The result is `3\n` I see.

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go (hunk 1 above)
Comment (wzhanw) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:1228]: oh, that should be a mistake,  I copied it from another name.

Comment (wzhanw) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go]: I corrected it and remove 1

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go (hunk 2 above)
Comment (ScrapCodes) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:831]: These are `from` and `to` parameters of a loop, do you think they are generated as a result of running a bash script?

**Pull request merged**